hathor/transaction/storage/transaction_storage.py

```diff
@@ -988,7 +998,40 @@ def _topological_fast(self) -> Iterator[BaseTransaction]:
         yield from cur_blocks
         yield from cur_txs
 
-    def _topological_sort(self) -> Iterator[BaseTransaction]:
+    def _topological_sort_metadata(self) -> Iterator[BaseTransaction]:
+        import heapq
+        from dataclasses import dataclass, field
+
+        @dataclass(order=True)
+        class Item:
+            timestamp: int
+            # XXX: because bools are ints, and False==0, True==1, is_transaction=False < is_transaction=True, which
+            #      will make blocks be prioritized over transactions with the same timestamp
+            is_transaction: bool
+            tx: BaseTransaction = field(compare=False)
+
+            def __init__(self, tx: BaseTransaction):
+                self.timestamp = tx.timestamp
+                self.is_transaction = tx.is_transaction
+                self.tx = tx
+
+        to_visit: List[Item] = list(map(Item, self.get_all_genesis()))
+        seen: Set[bytes] = set()
+        heapq.heapify(to_visit)
+        while to_visit:
+            item = heapq.heappop(to_visit)
+            assert item.tx.hash is not None
+            yield item.tx
+            # We can safely discard because no other tx will try to visit this one.
```

Comment:
This was not so obvious to me, I had to stop and think for a while about this.

Maybe if this comment explanation was expanded it could help some people (especially the ones not so used to the code/architecture)

Reply:
Done.